types.go

```diff
@@ -66,6 +70,10 @@ type SyncProgressListener interface {
 	OnSyncCanceled(willRestart bool)
 	OnSyncEndedWithError(err error)
 	Debug(debugInfo *DebugInfo)
+
+	OnTransaction(transaction string)
+	OnBlockAttached(walletID int, blockHeight int32)
+	OnTransactionConfirmed(walletID int, hash string, blockHeight int32)
```

Comment:
I think this should be methods on a different interface. Clients can register listeners for both interfaces in one file.

Reply:
Current dcrandroid implementation requires it this way and others will follow suit after implementing the new interface.

Reply:
dcrandroid usage should generally not dictate the design pattern; the reverse should be the case. We come up with a proper design pattern first then dcrandroid and other UI projects follow suit. These 3 notification methods are not related to an ongoing sync operation, so should go in a different interface. In dcrandroid, you can have one class implement both interfaces if you prefer.